tests/test_sqlite.py

```diff
@@ -1,34 +1,61 @@
-from ms2query.sqlite import create_table_structure, \
-    add_spectra_to_database, get_spectra_from_sqlite
-from ms2query.app_helpers import gather_test_json
+from ms2query.create_sqlite_database import make_sqlfile_wrapper
 import os
-import json
 from matchms.Spectrum import Spectrum
+from ms2query.query_from_sqlite_database import \
+    get_tanimoto_score_for_inchikeys, get_spectra_from_sqlite
+import pandas as pd
+from ms2query.app_helpers import load_pickled_file
 
 
 def test_sqlite_functions(tmp_path):
-    """Tests the functions create_table_structure, add_spectra_to_database
-    and get_spectra_from sqlite"""
-
-    column_type_dict = {"spectrum_id": "VARCHAR",
-                        "source_file": "VARCHAR",
-                        "ms_level": "INTEGER",
-                        "key_that_does_not_exist": "VARCHAR"
-                        }
-    sqlite_file_name = os.path.join(tmp_path, "test_spectra_database.sqlite")
-    spectra_id_list = ['CCMSLIB00000223876', 'CCMSLIB00003138082']
+    """Tests create_sqlite_database.py and query_from_sqlite_database.py
+    """
 
-    # Get path to json testfile
-    path_to_json_test_file = gather_test_json("testspectrum_library.json")[0][
-        "testspectrum_library.json"]
+    sqlite_file_name = os.path.join(tmp_path, "test_spectra_database.sqlite")
 
-    create_table_structure(sqlite_file_name, column_type_dict)
-    add_spectra_to_database(sqlite_file_name, path_to_json_test_file)
-    spectra_list = get_spectra_from_sqlite(sqlite_file_name, spectra_id_list)
+    # Create path so it also works for automatic testing
+    path_to_test_files_sqlite_dir = os.path.join(
+        os.path.split(os.path.dirname(__file__))[0],
+        'tests/test_files_sqlite')
+    # Create sqlite file, with 3 tables
+    make_sqlfile_wrapper(sqlite_file_name,
+                         os.path.join(path_to_test_files_sqlite_dir,
+                                      "test_tanimoto_scores.npy"),
+                         os.path.join(path_to_test_files_sqlite_dir,
+                                      "first_10_spectra.pickle"),
+                         os.path.join(path_to_test_files_sqlite_dir,
+                                      "test_metadata_for_inchikey_order.csv"))
 
     # Test if file is made
     assert os.path.isfile(sqlite_file_name), "Expected a file to be created"
 
+    tanimoto_score_dataframe = get_tanimoto_score_for_inchikeys(
+        ['MYHSVHWQEVDFQT',
+         'BKAWJIRCKVUVED',
+         'CXVGEDCSTKKODG'],
+        sqlite_file_name)
+
+    # The matrix in the testfile: [[1.0, 0.068, 0.106],
```

Comment:
This part made me realize that it is a bit of unexpected behavior to return the matrix with `NaN`s in there. So better to reconstruct the full array I think.
And then you could also test the entire values in addition, e.g. `assert np.allclose(tanimoto_score_dataframe.values, expected_array, atol=1e-6)`

Reply:
I implemented this and used a different test method: dataframe.equals(), since this also checks if the columns and indexes are correct.